core/services/ardupilot_manager/mavlink_proxy/test_all.py

```diff
@@ -106,29 +102,28 @@ def test_mavlink_router() -> None:
     assert mavlink_router.set_logdir(pathlib.Path(".")), "Local path as MAVLinkRouter log directory failed."
     assert re.search(r"\d+", str(mavlink_router.version())) is not None, "Version does not follow pattern."
 
-    endpoint_1 = Endpoint("udpout", "0.0.0.0", 14551)
-    endpoint_2 = Endpoint("tcpin", "0.0.0.0", 14552)
-    endpoint_3 = Endpoint("tcpout", "0.0.0.0", 14553)
-    assert mavlink_router.add_endpoint(endpoint_1), "Failed to add first endpoint"
-    assert mavlink_router.add_endpoint(endpoint_2), "Failed to add second endpoint"
-    assert mavlink_router.add_endpoint(endpoint_3), "Failed to add third endpoint"
-    assert mavlink_router.endpoints() == [
-        endpoint_1,
-        endpoint_2,
-        endpoint_3,
-    ], "Endpoint list does not match."
-    with pytest.raises(NotImplementedError):
-        mavlink_router.add_endpoint(Endpoint("udpin", "0.0.0.0", 14551))
-    with pytest.raises(NotImplementedError):
-        mavlink_router.add_endpoint(Endpoint("serial", "/dev/autopilot", 115200))
-
-    assert mavlink_router.start(Endpoint("udpin", "0.0.0.0", 14550)), "Failed to start MAVLinkRouter"
-    assert mavlink_router.is_running(), "MAVLinkRouter is not running after start."
-    assert mavlink_router.exit(), "MAVLinkRouter could not stop."
-    assert mavlink_router.start(Endpoint("serial", "/dev/autopilot", 115200)), "Failed to start MAVLinkRouter"
-    assert mavlink_router.is_running(), "MAVLinkRouter is not running after start."
-    assert mavlink_router.exit(), "MAVLinkRouter could not stop."
-    with pytest.raises(NotImplementedError):
-        mavlink_router.start(Endpoint("udpout", "0.0.0.0", 14550))
-    with pytest.raises(NotImplementedError):
-        mavlink_router.start(Endpoint("tcpout", "0.0.0.0", 14550))
+    allowed_output_types = [EndpointType.UDPClient, EndpointType.TCPServer, EndpointType.TCPClient]
+    allowed_master_types = [EndpointType.UDPServer, EndpointType.Serial, EndpointType.TCPServer]
+    run_common_routing_tests(mavlink_router, allowed_output_types, allowed_master_types)
+
+
+def run_common_routing_tests(
+    router: AbstractRouter, allowed_output_types: List[EndpointType], allowed_master_types: List[EndpointType]
+) -> None:
+    allowed_output_endpoints = [e for e in OUTPUT_ENDPOINTS if e.connection_type in allowed_output_types]
+    for endpoint in allowed_output_endpoints:
+        assert router.add_endpoint(endpoint), f"Failed to add endpoint {endpoint}."
+    assert router.endpoints() == allowed_output_endpoints, "Endpoint list does not match."
+
+    for endpoint in [e for e in OUTPUT_ENDPOINTS if not e.connection_type in allowed_output_types]:
```

Comment:
just use the `difference` function between between both sets.

Reply:
They are not sets yet, but since they are hashable now, they can be.

Reply:
Oh, wait, those are different types. The first one are the endpoints itself, while the second is a list of types allowed. I will implement the filter/lambda solution here.

Reply:
Got your point now.

Reply:
Making them set demands that we change it to sets on the AbstractRouter. I will open an issue for that and attack on a following PR.

Reply:
So, can't you just make a filter ?

Reply:
Yes yes, same way as the other.